.travis.yml

```diff
@@ -0,0 +1,178 @@
+#  - psql -U postgres -d odm2test -a -f ./tests/schemas/postgresql/ODM2_for_PostgreSQL.sql
+  - psql -c 'DROP DATABASE IF EXISTS odm2;' -U postgres;
+  - psql -c 'create database odm2;' -U postgres;
+  # patterned after: https://github.com/ptrv/gpx2spatialite/blob/master/.travis.yml
+  - sudo apt-get install -y python-software-properties
+  - sudo apt-add-repository -y ppa:git-core/ppa
+  - sudo apt-add-repository -y ppa:ubuntugis/ppa
+  - sudo apt-get update -qq
+  - sudo apt-get install unixodbc  unixodbc-dev odbcinst1debian2 odbcinst
+  - sudo apt-get install freetds-dev freetds-bin tdsodbc
+  - sudo apt-get install libc6  e2fsprogs  # mssql driver
+  # see what's installed
+ # - odbcinst -j
+  - ls /usr/lib/x86_64-linux-gnu/odbc/
+  - echo "[FreeTDS]"> .odbcinst.ini
+  - echo "Description = v0.91 with protocol v7.2">> .odbcinst.ini
+  - echo "Driver = /usr/lib/x86_64-linux-gnu/odbc/libtdsodbc.so">> .odbcinst.ini
+  - sudo odbcinst -i -d -f .odbcinst.ini
+  - echo "[azure]" > azure.odbc
+  - echo "Server = nrb8xkgxaj.database.windows.net" >> azure.odbc
+  - echo "Driver      = FreeTDS" >> azure.odbc
+  - echo "Database = odm2" >> azure.odbc
+  - sudo odbcinst -i -s -l -f azure.odbc
+  - echo "[moonstone]" > moonstone.odbc
+  - echo "Server = moonstone.sdsc.edu" >> moonstone.odbc
+  - echo "Driver      = FreeTDS" >> moonstone.odbc
+  - echo "Database = odm2" >> moonstone.odbc
+  - sudo odbcinst -i -s -l -f moonstone.odbc
+  # free tds
+  - sudo echo "[FreeTDS]" >> odbcinst.ini
+  - sudo echo "Description = v0.91 with protocol v7.2" >> odbcinst.ini
+  - sudo echo "Driver = /usr/lib/x86_64-linux-gnu/odbc/libtdsodbc.so " >> odbcinst.ini
+#  - sudo echo "Setup = $HOME/oob/client/libesoobsetup.so >> odbcinst.ini
+#  - sudo echo "FileUsage = 1" >> odbcinst.ini
+  #- ls -al /etc/odbcinst.ini /etc/odbc.ini
+  - cat /etc/odbcinst.ini
+  - cat  /etc/odbc.ini
+ # - ls /etc/ODBCDataSources
+ # - cat  .odbc.ini #not found
+  #
+  - echo "[global]" > .freetds.conf
+  - echo "tds version = 4.2" >> .freetds.conf
+  - echo "[kyle]" >> .freetds.conf
+  - echo "host = nrb8xkgxaj.database.windows.net" >> .freetds.conf
+  - echo "port = 1433" >> .freetds.conf
+  - echo "database=odm2" >> .freetds.conf
+  - echo "[moonstone]">> .freetds.conf
+  - echo "host = moonstone.sdsc.edu">> .freetds.conf
+  - echo "port = 1433">> .freetds.conf
+  - echo "database=odm2" >> .freetds.conf
+  - cat .freetds.conf
+  - sudo apt-get install mysql-client
+  # Spatialiate
+  - sudo apt-get install -y libproj-dev libgeos-dev libspatialite-dev
+  - sudo ln -s /usr/lib/x86_64-linux-gnu/libspatialite.so /usr/lib/libspatialite.so
+#  - sudo apt-get install python-scipy python-matplotlib python-pandas python-sympy python-nose
+  - sudo apt-get install python-matplotlib python-pandas python-nose
+  - python -m pip install pandas
+  - export PYMSSQL_BUILD_WITH_BUNDLED_FREETDS=1;python -m  pip install pymssql
+  - python -m  pip install mysql-python
```

Comment:
This is a very complex `before_install` maybe it is worth re-factoring it into scripts with specific goals like: `create_db.sh`, etc.